legacy/signals/signals.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signals
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+// sigMap holds all signals the Watch() function should terminate
+// when encountering. The key holds the string representation
+// of the captured os.Signal. The value is the actual syscall of the
+// signal which allows the Watch() function to terminate with that exit code.
+var sigMap = map[string]syscall.Signal{
+	"hangup":              syscall.SIGHUP,
+	"interrupt":           syscall.SIGINT,
+	"aborted":             syscall.SIGABRT,
+	"illegal instruction": syscall.SIGILL,
+	"quit":                syscall.SIGQUIT,
+	"terminated":          syscall.SIGTERM,
+	"segmentation fault":  syscall.SIGSEGV,
+	"stopped":             syscall.SIGTSTP,
+}
+
+// Watch concurrenlty logs debug statements when encountering interrupt signals from the OS.
+// This function relies on the os/signal package which may not capture SIGKILL and SIGSTOP.
+func Watch() {
+	c := make(chan os.Signal, 1)
+	// Observe all signals, excluding SIGKILL and SIGSTOP.
+	signal.Notify(c)
+	// Continuously watch for signals.
+	go func() {
+		logrus.Debug("Watching for OS Signals...")
+		for sig := range c {
+			logrus.Debug("Encoutered signal: ", sig.String())
+			// Ensure the current program stops when encountering
+			// an interrupt. Without this check, these signals would
+			// not cause the program to terminate.
+			if s, exists := sigMap[sig.String()]; exists {
```

Comment:
Instead of using `sig.String()` as the key, why not `sig.Signal()` ?

Reply:
[sig.Signal() ](https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/os/exec.go;l=69-72;bpv=1;bpt=1)does not return anything. For this reason, I use sig.String().

Reply:
Then I guess we can just use `int(s)` instead?

Reply:
sig is of type os.Signal which is an interface. I must use [syscall.Signal](https://pkg.go.dev/syscall#Signal)